atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -225,6 +226,7 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         } finally {
             postTaskContext = postTaskTimer.time();
             timelockService.tryUnlock(ImmutableSet.of(txAndLock.immutableTsLock()));
+            lockWatchService.removeTimestampFromCache(tx.getTimestamp());
```

Comment:
Since we don't have the `CommitUpdate` / `PreCommitCondition` piece done, timestamps need to be removed from the cache at some point.